msteams-platform/m365-apps/actions-in-m365.md

```diff
@@ -0,0 +1,213 @@
+
+When you're creating an app ensure that you define user intent, choose the object to perform the action, and construct the corresponding handler that facilitates task completion for the user.
+
+### Intent
+
+Intent is the objective a user wants to perform or achieve, such as open or add. Microsoft 365 uses intent to display Actions in locations that align with the user’s needs and intentions. Intent determines the placement, grouping, and ordering of Actions. You can create an intent for Open, addTo, and custom actions. You can use custom Actions to create tailored Actions.
+
+### Object
+
+Object is the file on which the user wants to perform an action. Currently, Actions can be triggered on content objects (files) that have an extension, such as Word, PowerPoint, Excel, PDF, and images. The files must be available in OneDrive or SharePoint and are accessible through Microsoft Graph.
+
+### Handlers
+
+A handler is how the Action performs the user’s intent on the selected object. It provides the logic and functionality of the Action, creating a smooth and meaningful user experience.
+
+The following are the available types of handlers:
+
+* `openDialog`: Directs users to a dialog, offering a dedicated and contextualized interface for interacting with your app's features without opening the full app.
+* `openPage`: Drive users to your app's dedicated pages (personal tab).
+
+### Prerequisites
+
+Ensure you install the following tools to build Actions in Microsoft 365:
+
+| &nbsp; | Install | For using... |
+| --- | --- | --- |
+| &nbsp; | [Teams Toolkit](../toolkit/install-Teams-Toolkit.md) | A Microsoft Visual Studio Code extension that creates a project scaffolding for your app. Use the latest version. |
+| &nbsp; | [Microsoft Teams](https://www.microsoft.com/microsoft-teams/download-app) | Microsoft Teams to collaborate with everyone you work with through apps for chat, meetings, and call-all in one place.|
+| &nbsp; | [Node.js](https://nodejs.org/en/download/) | Back-end JavaScript runtime environment. For more information, see [Node.js version compatibility table for project type](~/toolkit/build-environments.md#nodejs-version-compatibility-table-for-project-type).|
+| &nbsp; | [Microsoft Edge](https://www.microsoft.com/edge) (recommended) or [Google Chrome](https://www.google.com/chrome/) | A browser with developer tools. |
+| &nbsp; | [Visual Studio Code](https://code.visualstudio.com/download) | JavaScript, TypeScript, or SharePoint Framework (SPFx) build environments. Use the latest version. |
+
+### Create Actions
+
+To create Actions for your app, follow these steps:
+
+1. [Update app manifest](#update-app-manifest).
+1. [Create a handler](#create-a-handler).
+1. [Access content through Graph API](#access-content-through-graph-api).
+
+#### Update app manifest
+
+Define the intent, object, and handler for your actions in the app manifest schema (previously called Teams app manifest).
```

Comment:
Could we reinstate a similar description from original private preview article, to clarify to the reader the purpose of this step:

To make an action show up in the context menu when the user right-clicks on a piece of content in Microsoft 365 app,  simply add a json payload in the manifest, where you define the user intent, object type and the handler that will trigger.